cloud/terraform/terraform_controller.py

```diff
@@ -0,0 +1,105 @@
+from ast import alias
+import os
+import json
+
+
+class TerraformController:
+    def __init__(self, cloud):
+        self.cloud = cloud
+
+        if cloud == 'aws':
+            self.cloud_instance = 'aws_instance'
+        self.resources = {'resource': {self.cloud_instance: {}}}
+        self.providers = {'provider': {cloud: []}}
+
+    def __del__(self):
+        self.destroy_infra()
+
+    def __dump_to_json(self, content, file):
+        with open(file, 'w') as config_file:
+            json.dump(content, config_file)
+
+    def set_aws_profile(self, profile):
+        self.aws_profile = profile
+
+    def add_region(self, region):
+        if self.cloud == 'aws':
+            # If no profile was specified, select the default
+            if not hasattr(self, 'aws_profile'):
+                self.aws_profile = 'aws'
+
+            new_provider = {'region': region, 'alias': region, 'profile': self.aws_profile}
+
+        self.providers['provider'][self.cloud].append(new_provider)
+
+    def add_instance(self, region, image_id, instance_type=''):
+        aliases = [provider["alias"] for provider in self.providers["provider"][self.cloud]]
+        if not region in aliases:
```

Comment:
Isn't it `if region not in aliases`? Maybe both are valid, but for me it looks more natural to read this way. What do you think?